java/src/jmri/jmrit/display/IndicatorTurnoutIcon.java

```diff
@@ -413,8 +416,10 @@ public void propertyChange(java.beans.PropertyChangeEvent evt) {
     private void setStatus(OBlock block, int state) {
         _status = _pathUtil.getStatus(block, state);
         if ((state & (OBlock.OCCUPIED | OBlock.RUNNING)) != 0) {
-            _pathUtil.setLocoIcon(block, getLocation(), getSize(), _editor);
-            repaint();
+            ThreadingUtil.runOnLayoutEventually(() -> {
```

Comment:
It would be better to use `ThreadingUtil.runOnGUIEventually()`. The layout thread is currently the same as the GUI thread, but in a distant future, they could be separated.